### PR TITLE
Add IAsyncDisposable interface support

### PR DESCRIPTION
**Pull request by halex2005**
Greetings.

I'd like to have async disposable pattern with async lock. 
Here is simplest implementation, please consider to review and merge it.

### CODE REVIEW

File: Vostok.ZooKeeper.Recipes.Tests/DistributedLock_Tests.cs [hunk 1]
```diff
@@ -38,6 +38,22 @@ public async Task AcquireAsync_should_work_alone()
             CheckNoLocks();
         }
 
+#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_0_OR_GREATER
```
Comment (ingevarr) [inline Vostok.ZooKeeper.Recipes.Tests/DistributedLock_Tests.cs:41]: Redundant target framework directives. You can use #if NET6_0_OR_GREATER only

File: Vostok.ZooKeeper.Recipes/Helpers/DistributedLockToken.cs [hunk 2]
```diff
@@ -70,5 +70,29 @@ public void Dispose()
                 delete.EnsureSuccess();
             }
         }
+        
+#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_0_OR_GREATER
```
Comment (ingevarr) [inline Vostok.ZooKeeper.Recipes/Helpers/DistributedLockToken.cs:74]: Redundant target framework directives. You can use #if NET6_0_OR_GREATER only

File: Vostok.ZooKeeper.Recipes/IDistributedLockToken.cs [hunk 3]
```diff
@@ -10,7 +10,11 @@ namespace Vostok.ZooKeeper.Recipes
     /// <para>Should be disposed in order to release the lock.</para>
     /// </summary>
     [PublicAPI]
-    public interface IDistributedLockToken : IDisposable
+    public interface IDistributedLockToken
+    : IDisposable
+    #if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_0_OR_GREATER
```
Comment (ingevarr) [inline Vostok.ZooKeeper.Recipes/IDistributedLockToken.cs:15]: Redundant target framework directives. You can use #if NET6_0_OR_GREATER only

File: Vostok.ZooKeeper.Recipes/Helpers/DistributedLockToken.cs [hunk 4]
```diff
@@ -70,5 +70,29 @@ public void Dispose()
                 delete.EnsureSuccess();
             }
         }
+        
+#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_0_OR_GREATER
+        public async ValueTask DisposeAsync()
+        {
+            if (disposed.TrySetTrue())
```
Comment (ingevarr) [inline Vostok.ZooKeeper.Recipes/Helpers/DistributedLockToken.cs:77]: Code fully duplicated from Dispose(). It's better to extract method and reuse it both in Dispose and DisposeAsync

**Review by yngevarr — CHANGES_REQUESTED** [review]
Greetings! Need some small changes

**Pull request closed without merging**